File: Python/Recursion-1/pair_star.py

# given a string, compute recursively a new string where identical chars that are adjacent in the original string are separated from each other by a "*"
def pair_star(str):
    s = ''

    if not str:
        return s
    if len(str) > 1 and str[0] == str[1]:
        return s + str[0] + '*' + pair_star(str[1:])
    else:
        return s + str[0] + pair_star(str[1:])

    return 0

print(pair_star("hello"))
print(pair_star("xxyy"))
print(pair_star("aaaa"))

File: Java/Recursion-1/pairStar.java

public class recursion1 {
	
	// given a string, compute recursively a new string where identical chars that are adjacent in the original string are separated from each other by a "*"
	public String pairStar(String str) {
	
		String s = "";
				
		if (str.isEmpty()) {
			return s;
		}
		if (str.length() > 1 && str.charAt(0) == str.charAt(1)) {
			return s + str.charAt(0) + "*" + pairStar(str.substring(1));
		}
		else {
			return s + str.charAt(0) + pairStar(str.substring(1));
		}
		
	}

	public static void main(String[] args) {

		recursion1 o = new recursion1();
		
		System.out.println(o.pairStar("hello"));
		System.out.println(o.pairStar("xxyy"));
		System.out.println(o.pairStar("aaaa"));

	}

}
